test/Table_Tests/src/Common_Table_Operations/Derived_Columns_Spec.enso

```diff
@@ -229,6 +237,26 @@ spec setup =
             t2 = t.set (Column_Operation.Add "prefix" (Column_Ref.Name "X"))
             t3 = t2.set (Column_Operation.Add "prefix" "X")
 
-            t3.column_names . should_equal ["X", "['prefix'] + [X]", "['prefix'] + 'X'"]
-            t3.at "['prefix'] + [X]" . to_vector . should_equal ["prefixa", "prefixb", "prefixc"]
-            t3.at "['prefix'] + 'X'" . to_vector . should_equal ["prefixX", "prefixX", "prefixX"]
+            t3.column_names . should_equal ["X", "[prefix] + [X]", "[prefix] + 'X'"]
+            t3.at "[prefix] + [X]" . to_vector . should_equal ["prefixa", "prefixb", "prefixc"]
+            t3.at "[prefix] + 'X'" . to_vector . should_equal ["prefixX", "prefixX", "prefixX"]
+
+        Test.specify "Should not disambiguate if set_mode is Update" <|
+            t = table_builder [["X", [1, 2, 3]]]
+            t2 = t.set (Column_Operation.Add (Column_Ref.Name "X") 1) set_mode=Set_Mode.Update
+            t2.column_names . should_equal ["X"]
+            t2.at "X" . to_vector . should_equal [2, 3, 4]
+
+        Test.specify "Should not disambiguate if set_mode is Add_Or_Update" <|
+            t = table_builder [["X", [1, 2, 3]], ["[X] + 1", [10, 20, 30]]]
+            # set_mode=Set_Mode.Add_Or_Update is the default
+            t2 = t.set (Column_Operation.Add (Column_Ref.Name "X") 1)
+            t2.column_names . should_equal ["X", "[X] + 1"]
+            t2.at "X" . to_vector . should_equal [1, 2, 3]
+            t2.at "[X] + 1" . to_vector . should_equal [2, 3, 4]
```

Comment:
This looks good.

But, I think I don't understand - why is in one case the new column name just kept as `X` (if Update) but otherwise (if Add_Or_Update) it is `[X] + 1`.
I cannot find the condition in the code that is driving this change.

Reply:
Ah okay, it is the `column.evaluate self (set_mode==Set_Mode.Update && new_name=="") on_problems` that was there before.

Yeah I think all looks good.

Reply:
I'm a little worried that the logic for when the column name is 'the name of the base column' and when it is 'the text of the expression used to create it' is not that clear to the user.

If there is not many columns in the table - it is not a big deal - the user will usually see where the new column was created.

I'm worried however that if there are many columns, it may be unclear for the user what name was picked for the column - and if it possible overwrote an existing column or not.

---

But of course all of this is solved if the user explicitly sets `Update` or `Add` instead of `Add_Or_Update` and sets an explicit `new_name` - then all is clear.

So I guess the behaviour is OK - the defaults seem rather sane and intuitive. I'm worried they may start to confuse in more complex scenarios, but I don't see much better alternatives. Overall, I think this is good.

--
And the test coverage is really nice 🙂

Reply:
I'm concerned about it too; perhaps Set_Mode should work differently for derived columns, vs. columns that are named by the user or from data sources. But we can see how things work before changing it.